Expose hue group 0

filcole opened the pull request:
## Description:


**Related issue (if applicable):** fixes #8652

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml
light:
  platform: hue
  host: <hueip>
  allow_hue_groups: true
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

filcole committed:
Tado Fix #8606

Handle case where 'mode' and 'fanSpeed' are missing JSON. Based on
changes in commit
https://github.com/wmalgadey/tado_component/commit/adfb608f86b8bf4c1c43e71b4067cbfe1de9ba85

filcole committed:
Merge remote-tracking branch 'refs/remotes/home-assistant/dev' into dev

filcole committed:
Expose hue group 0 to HA #8652

If allow_hue_groups is set expose "All Hue Lights" group for "special
group 0".  This does add an additional Hue API call for every refresh
(approx 30 secs) to get the status of the special group 0 because it's
not included in the full API pull that currently occurs.

filcole committed:
Revert "Expose hue group 0 to HA #8652"

This reverts commit db7fe47ec72a4907f8a59ebfb47bc4a6dfa41e89.

filcole committed:
Expose hue group 0 to HA #8652

If allow_hue_groups is set expose "All Hue Lights" group for "special
group 0".  This does add an additional Hue API call for every refresh
(approx 30 secs) to get the status of the special group 0 because it's
not included in the full API pull that currently occurs.

danielfaulknor commented:
I'm not entirely convinced that all_lamps is a good name? For example if you have another light system as well as hue, it won't include those. all_hue_lamps perhaps? or even just hue_lamps?

balloob commented on homeassistant/components/light/hue.py:
```diff
@@ -205,6 +205,18 @@ def update_lights():
+            # Group ID 0 is a special group in the hub for all lights, but it's not 
+            # returned by get_api() so explicity get it and include it. See
+            # https://developers.meethue.com/documentation/groups-api#21_get_all_groups
+            all_lamps = bridge.get_group(0)
```
Please rename to `all_lights` to stay consistent with rest of variables.

balloob commented on homeassistant/components/light/hue.py:
```diff
@@ -205,6 +205,18 @@ def update_lights():
+                _LOGGER.error("Got unexpected result from Hue API for group 0")
+                return
+            # Hue hub returns name of group 0 as "Group 0", so rename for ease of use in HA
+            all_lamps['name'] = "All Hue Lights" 
```
Please extract this as a constant.

balloob reviewed (changes requested):
Nice change.

You'll also need to fix the errors by hound / reported by Travis (click details to the right of build failed)

andrey-git commented:
@filcole any updates on this PR?

filcole commented:
Plan to work on this this week.

filcole committed:
Changes per review by balloob

1) Use all_lights instead of all_lamps
2) Fix line lengths and trailing whitespace
3) Move "All Hue Lights" to GROUP_NAME_ALL_HUE_LIGHTS constant

filcole committed:
Make "All Hue Lights" a constant

filcole committed:
Fix trailing whitespace

filcole commented:
I think this is good to go now.  Let me know if there's any issues.  Thanks.

balloob commented:
Thanks! :dolphin:
